docs/non-normative/k8s-migration.md

```diff
@@ -0,0 +1,65 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: K8s migration
+--->
+
+# K8s semantic conventions stability migration
+
+Due to the significant number of modifications and the extensive user base
+affected by them, existing K8s instrumentations published by
+OpenTelemetry are required to implement a migration plan that will assist users in
+transitioning to the stable K8s semantic conventions.
+
+When existing K8s instrumentations published by OpenTelemetry are
+updated to the stable K8s semantic conventions, they:
+
+- SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` in
+  their existing major version, which accepts:
+  - `k8s` - emit the stable k8s conventions, and stop emitting
+    the old k8s conventions that the instrumentation emitted previously.
+  - `k8s/dup` - emit both the old and the stable k8s conventions,
+    allowing for a phased rollout of the stable semantic conventions.
+  - The default behavior (in the absence of one of these values) is to continue
+    emitting whatever version of the old k8s conventions the
+    instrumentation was emitting previously.
+- Need to maintain (security patching at a minimum) their existing major version
+  for at least six months after it starts emitting both sets of conventions.
+- May drop the environment variable in their next major version and emit only
+  the stable k8s conventions.
+
+Specifically for the Opentelemetry Collector:
+
+The transition will happen through two different feature gates.
+One for enabling the new schema called `semconv.k8s.stable`,
+and one for disabling the old schema called `semconv.k8s.legacy`. Then:
+
+- On alpha the old schema is enabled, while the new schema is disabled
+- On beta/stable the old schema is disabled, while the new is enabled
+- It is an error to disable both
```

Comment:
similar to `k8s.dup`, will users be allows to set both to true to emit both semantics?

Reply:
Yes, I applied a change to the feature gates' names and enhanced the description to make it clear.

The idea is:

```
- On alpha the old schema is enabled by default (`semconv.k8s.disableLegacy` defaults to false), 
  while the new schema is disabled (`semconv.k8s.enableStable` defaults to false).
- On beta/stable the old schema is disabled (`semconv.k8s.disableLegacy` defaults to true),
  while the new is enabled (`semconv.k8s.enableStable` defaults to true).
```

To have both schemas (`k8s/dup`) we need to configure:
- On alpha: `otelcol --config=config.yaml --feature-gates=semconv.k8s.enableStable`
- On Beta:  `otelcol --config=config.yaml --feature-gates=semconv.k8s.disableLegacy`

@TylerHelmuth @mx-psi would that make sense? (I'm always confused with the feature gates' life-cycle, please correct me here if I'm wrong :))

Reply:
To enable both the old and new set of metrics you would have to pass 
```
--config=config.yaml --feature-gates=-semconv.disableLegacy,+semconv.k8s.enableStable
```

Both feature gates would graduate from alpha to beta on the same release, and from beta to stable at the same time as well.

Reply:
Clarified that in place.

Reply:
Resolving this for now. We can continue at https://github.com/open-telemetry/semantic-conventions/pull/1597#discussion_r1866107784